app/src/components/progress-circle/ProgressCircle.tsx

```diff
@@ -0,0 +1,73 @@
+import { ProgressBar } from "react-aria-components";
+import { css } from "@emotion/react";
+
+import {
+  CENTER,
+  CIRCUMFERENCE,
+  progressCircleIndeterminateCSS,
+  RADIUS,
+  SIZES,
+  STROKE_WIDTHS,
+} from "./styles";
+import type { ProgressCircleProps } from "./types";
+
+function ProgressCircle({
```

Comment:
might want to just start with ref forwarding for various use-casees https://github.com/adobe/react-spectrum/blob/805fa34d8fde985de15be4e2d0487d88fe3fa61c/packages/%40react-spectrum/progress/src/ProgressCircle.tsx#L26